src/estimators/bloom_filters.py

```diff
@@ -383,6 +383,21 @@ def first_moment(u):
 
     return invert_monotonic(first_moment, lower_bound)(0)
 
+  @classmethod
+  def _estimate_cardinality_geo(cls, sketch, weights):
```

Comment:
@shengma-1 @huangxichen1 

I didn't review this for correctness of estimator, can you guys make sure you do?

Reply:
The reason I add this is to add a boundary check for denoised bloom filter to avoid the crash since in the original estimator, `1 - np.average(sketch.sketch) ` may be negative and will crash during the evaluation. I also optimized away `weights` since they're always 1. Other than this, there shouldn't be any differences